pkg/dns/dns.go

```diff
@@ -268,6 +301,124 @@ func (kd *KubeDNS) updateService(oldObj, newObj interface{}) {
 	kd.newService(newObj)
 }
 
+func assertIsIngress(obj interface{}) (*ext.Ingress, bool) {
+	if ing, ok := obj.(*ext.Ingress); ok {
+		return ing, ok
+	} else {
+		glog.Errorf("Type assertion failed! Expected object of type \"Ingress\", got %T", obj)
+		return nil, ok
+	}
+}
+
+// newIngress generates DNS records for an Ingress resource.
+//
+// If all the Ingress loadbalancer status entries have their IP fields
+// set we generate a series of A records for them, one per IP. If there
+// is only one hostname in the Ingress status entries, we generate a
+// CNAME record for it. If there are multiple hostnames and no IP then
+// we generate a CNAME for the first hostname. If both IPs and hostnames
+// coexist, we favor IPs over the hostnames and we generate only A
+// records for the IPs and no CNAME record at all.
+//
+// We do this because RFC 1034 - https://tools.ietf.org/html/rfc1034,
+// specifies that it is not a valid configuration to mix up CNAME and
+// A records for a given domain name.
+// """
+// If a CNAME RR is present at a node, no other data should be
+// present; this ensures that the data for a canonical name and its
+// aliases cannot be different.
+// """
+// Due to this, we can either only generate A records or a CNAME
+// record but not both. We have to choose between the two. IP addresses
+// are more specific than a CNAME because a CNAME should eventually
+// resolve to an IP address, so favoring IP addresses over CNAME.
+//
+// TODO(madhusudancs): We would like to generate an empty record
```

Comment:
@girishkalele There isn't. @nikhiljindal also pointed that out. I don't think it is that important enough to be tracked in an issue. But if you both insist, I will open an issue.